modules/bioinspired/src/transientareassegmentationmodule.cpp

```diff
@@ -454,7 +454,7 @@ void TransientAreasSegmentationModuleImpl::_run(const std::valarray<float> &inpu
 
     // first square the input in order to increase the signal to noise ratio
     // get motion local energy
-    _squaringSpatiotemporalLPfilter(&inputToSegment[channelIndex*getNBpixels()], &_localMotion[0]);
+    _squaringSpatiotemporalLPfilter(&const_cast<std::valarray<float>&>(inputToSegment)[channelIndex*getNBpixels()], &_localMotion[0]);
```

Comment:
I'm sure this would work as it would inform the compiler of the type of reference being taken - however, I feel like a dummy variable is more likely to end up getting accidentally refactored back out? However, if you want me to change it to that I can, just let me know.